tests/plugins/test_push_floating_tags.py

```diff
@@ -0,0 +1,610 @@
+        if not ref.startswith('sha256:'):
+            try:
+                ref = repo['tags'][ref]
+            except KeyError:
+                return (requests.codes.NOT_FOUND, {}, b"{'error': 'NOT_FOUND'}")
+
+        try:
+            blob = repo['manifests'][ref]
+        except KeyError:
+            return (requests.codes.NOT_FOUND, {}, {'error': 'NOT_FOUND'})
+
+        decoded = json.loads(to_text(blob))
+        content_type = decoded['mediaType']
+
+        accepts = re.split(r'\s*,\s*', req.headers['Accept'])
+        assert content_type in accepts
+
+        headers = {
+            'Docker-Content-Digest': ref,
+            'Content-Type': content_type,
+            'Content-Length': str(len(blob)),
+        }
+        return (200, headers, blob)
+
+    def _put_manifest(self, req, name, ref):
+        try:
+            json.loads(to_text(req.body))
+        except ValueError:
+            return (400, {}, {'error': 'BAD_MANIFEST'})
+
+        self.add_manifest(name, ref, req.body)
+        return (200, {}, '')
+
+    def _get_blob(self, req, name, digest):
+        repo = self.get_repo(name)
+        assert digest.startswith('sha256:')
+
+        try:
+            blob = repo['blobs'][digest]
+        except KeyError:
+            return (requests.codes.NOT_FOUND, {}, {'error': 'NOT_FOUND'})
+
+        headers = {
+            'Docker-Content-Digest': digest,
+            'Content-Type': 'application/json',
+            'Content-Length': str(len(blob)),
+        }
+        return (200, headers, blob)
+
+    def _mount_blob(self, req, target_name, digest, source_name):
+        source_repo = self.get_repo(source_name)
+        target_repo = self.get_repo(target_name)
+
+        try:
+            target_repo['blobs'][digest] = source_repo['blobs'][digest]
+            headers = {
+                'Location': '/v2/{}/blobs/{}'.format(target_name, digest),
+                'Docker-Content-Digest': digest,
+            }
+            return (201, headers, '')
+        except KeyError:
+            headers = {
+                'Location': '/v2/{}/blobs/uploads/some-uuid'.format(target_name),
+                'Docker-Upload-UUID': 'some-uuid',
+            }
+            return (202, headers, '')
+
+
+def mock_registries(registries, config, schema_version='v2'):
+    """
+    Creates MockRegistries objects and fills them in based on config, which specifies
+    which registries should be prefilled (as if by workers) with platform-specific
+    manifests, and with what tags.
+    """
+    reg_map = {}
+    for reg in registries:
+        reg_map[reg] = MockRegistry(reg)
+
+    worker_builds = {}
+
+    for platform, regs in config.items():
+        digests = []
+
+        for reg, tags in regs.items():
+            registry = reg_map[reg]
+            layer_digest = make_digest('layer-' + platform)
+            config_digest = make_digest('config-' + platform)
+
+            if schema_version == 'v2':
```

Comment:
Only v2 is supported for push_floating_tags right now. We're not creating any OCI image index object.
```suggestion
            assert schema_version == 'v2'
```